### PR TITLE
#159291136 Validate all cases of user's entries via Testing

### PR DESCRIPTION
**Pull request by phemmelliot**
#### What does this PR do?
    In this PR i wrote test for all the endpoints, the test was written with mocha 
   and chai and the test has support for es6. It has babel compilers add to the test script.
#### List of Task to be completed?
    1. installed mocha, chai and chai-http dependencies for testing
    2. Imported the dependencies into test file and wrote the test for all
        5 endpoints covering as much edge cases as possible.
    3. Included babel compilers in the test script 
    4. I added node_modules to gitignore
#### How should this be manually tested?
     This test can be carried out by using npm test
#### Any background context you want to provide?
     I was initially using jasmine to write my test, but it turns out to be a little hard 
     for me so changed to mocha and chai, then i was able to test my endpoints easily
#### What are the relevant pivotal tracker stories?
     Here is the link to 
   [pivotal tracker board](https://www.pivotaltracker.com/story/show/159291136)
#### Screenshot
     
test

### CODE REVIEW

File: node_modules/rimraf/package.json [hunk 1]
```diff
@@ -1,27 +1,29 @@
 {
-  "_from": "rimraf@^2.2.8",
+  "_from": "rimraf",
```
Comment (andela-ookoro) [inline node_modules/rimraf/package.json]: why did you post this file

File: package.json [hunk 2]
```diff
@@ -26,18 +27,31 @@
   },
   "homepage": "https://github.com/phemmelliot/Andela-Developer-Challenge#readme",
   "dependencies": {
+    "babel-core": "^6.26.3",
     "body-parser": "^1.18.3",
+    "chai": "^4.1.2",
+    "chai-http": "^4.0.0",
     "express": "^4.16.3",
+    "mocha": "^5.2.0",
+    "morgan": "^1.9.0",
```
Comment (andela-ookoro) [inline package.json]: what did you use `morgan` for

File: test/login.spec.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import { server, entries } from '../server';
```
Comment (andela-ookoro) [inline test/login.spec.js]: nice test

File: package.json (hunk 2 above)
Comment (phemmelliot) [reply package.json]: I was initially using a tutorial where morgan was used apparently for logging switching between config files of test and live mongo db, but because i did not use any config file, i didn't use morgan

File: test/login.spec.js (hunk 3 above)
Comment (phemmelliot) [reply test/login.spec.js]: Thank you

File: node_modules/rimraf/package.json (hunk 1 above)
Comment (phemmelliot) [reply node_modules/rimraf/package.json]: When i was going through babel issues yesterday, rimraf was added to build script for performing a unix command rm -rf on a folder, but it turns out to be an ineffective approach so i dumped, but i didn't know it was pushed, next i am pushing i will add node_modules to gitignore

**Review by andela-ookoro — APPROVED** [review]
this is good
